transformers4rec/torch/model/base.py

```diff
@@ -400,9 +403,20 @@ def forward(
             outputs = {"loss": loss, "labels": labels, "predictions": predictions}
         else:
             for name, task in self.prediction_task_dict.items():
-                outputs[name] = task(
-                    body_outputs, targets=targets, training=training, testing=testing, **kwargs
-                )
+                if isinstance(task, NextItemPredictionTask):
```

Comment:
If the `NextItemPredictionTask` is always something that is created by user code. An alternative to passing to the foward method could be accepting `top_k` in the constructor `__init__` method. That way we wouldn't need to have this condition here

Reply:
The reason we did not add this to the constructor is bcs there is no turning back changing the topK value at the inf step, once the model is trained with top_k arg. So to make it flexible, we decided to add it in the forward method.
```
head = tr.Head(
    body,
    tr.NextItemPredictionTask(weight_tying=True, 
                              metrics=metrics, top_k =20),
    inputs=inputs,
)
```

@sararb wanna add something else?